core/src/main/java/hudson/model/TimeZoneProperty.java

```diff
@@ -0,0 +1,111 @@
+package hudson.model;
+
+import hudson.Extension;
+import hudson.util.ListBoxModel;
+import hudson.util.ListBoxModel.Option;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.TimeZone;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+/**
+ * A UserProperty that allows a user to specify a time zone for displaying time.
+ *
+ * @author Nathan Grubb
+ */
+public class TimeZoneProperty extends UserProperty implements Saveable {
+    /**
+     * Time Zone ID defined by the user.
+     * {@code null} means that the time zone is not defined.
+     */
+    @CheckForNull
+    private String timeZoneName;
+
+    private static final Logger LOGGER = Logger.getLogger(TimeZoneProperty.class.getName());
+
+    @DataBoundConstructor
+    public TimeZoneProperty(@CheckForNull String timeZoneName) {
+        this.timeZoneName = timeZoneName;
```

Comment:
So I have a few alternative ways you could keep the time zone valid without having to use that ugly workaround.

1. Modify this constructor to check if the timezone is valid; if not, throw an `IllegalArgumentException`.
2. Not sure if this works, but change the type of the field, constructor argument, and getter, to `TimeZone`, and then add a `writeReplace()`/`ProxyObject::readResolve()` pair to serialize it since `java.util.TimeZone` is not whitelisted for serialization.
3. Alternatively, maybe we can add `java.util.TimeZone` to the whitelisted classes for serialization combined with the first part of 2.

Reply:
It's not about keeping it valid, it's already not possible to be invalid but this is error handling if for some reason it was changed through configuration as code. This checks if it's valid and is the cleanest and most efficient way I think to check if it's valid.

Reply:
Oh, I see. Then I think adding the validation in the constructor here would make more sense than trying to recover from a failure that shouldn't happen in practice (except for typos in hand-written config as code, but that's always a given with yaml files).

Reply:
Also your ideas run counter to your coworkers suggestion. I give up, this is all honestly quite frustrating and I don't want to do this in my free time. I'm going to investigate using TeamCity or just living with Jenkins as is.

Reply:
Coworkers disagree; what else is new? This same debate would be had if we all worked at different companies. ;)

Reply:
Fair, but just difficult being on the outside as to what the actual guideline is for handling errors in Jenkins, but I'm guessing like most open source there's no particular standard.